packages/hardhat/contracts/periphery/ControllerHelper.sol

```diff
@@ -162,12 +177,66 @@ contract ControllerHelper is FlashControllerHelper, IERC721Receiver {
             _wPowerPerpAmountToBurn.add(_wPowerPerpAmountToBuy),
             _collateralToBuyWith.add(msg.value),
             uint8(FLASH_SOURCE.FLASH_W_BURN),
-            abi.encodePacked(_vaultId, _wPowerPerpAmountToBurn, _wPowerPerpAmountToBuy, _collateralToWithdraw, _collateralToBuyWith.add(msg.value))
+            abi.encodePacked(
+                _vaultId,
+                _wPowerPerpAmountToBurn,
+                _wPowerPerpAmountToBuy,
+                _collateralToWithdraw,
+                _collateralToBuyWith.add(msg.value)
+            )
         );
 
         emit FlashWBurn(msg.sender, _vaultId, _wPowerPerpAmountToBurn, _collateralToWithdraw, _wPowerPerpAmountToBuy);
     }
 
+    /**
+     * @notice mint WPowerPerp and LP into Uniswap v3 pool
+     * @param _vaultId vault ID
+     * @param _wPowerPerpAmount amount of WPowerPerp token to mint
+     * @param _collateralToMint collateral to use for minting
+     * @param _collateralToLP collateral to use for LPing
+     * @param _lowerTick LP lower tick
+     * @param _upperTick LP upper tick
+     */
+    function batchMintLp(
+        uint256 _vaultId,
+        uint256 _wPowerPerpAmount,
+        uint256 _collateralToMint,
+        uint256 _collateralToLP,
+        int24 _lowerTick,
+        int24 _upperTick
+    ) external payable {
+        require(msg.value == _collateralToMint.add(_collateralToLP), "Wrong ETH sent");
+
+        uint256 vaultId = IController(controller).mintWPowerPerpAmount{value: _collateralToMint}(
+            _vaultId,
+            _wPowerPerpAmount,
+            0
+        );
+        address token0 = IUniswapV3Pool(wPowerPerpPool).token0();
```

Comment:
cheaper to store on contract creation vs reading every time

Reply:
see this? am I correct, shouldn't we store this on contract creation in constructor vs calling it externally for every use?